scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1624,56 +1671,95 @@
-          considerable-job-uuids (set (map :job/uuid considerable-jobs))]
-      (log-structured/info "Considering jobs" {:pool pool-name :number-considered-jobs (count considerable-job-uuids)})
-      (if (seq considerable-jobs)
+    (let [; This is required to properly filter compute clusters to
+          ; those configured for this pool. This is done via the synthetic-pods
+          ; config, accessed via the autoscaling? method. In the future, this list
+          ; of pools should be slightly restructured to be listed one level higher,
+          ; on the compute-cluster-template config.
+          compute-clusters-for-pool (filter #(cc/autoscaling? % pool-name) compute-clusters) 
+          max-total-considerable (:max-jobs-considered scheduler-config)
+          minimum-scheduling-capacity-threshold (:minimum-scheduling-capacity-threshold scheduler-config)
+          scheduling-pause-time-ms (:scheduling-pause-time-ms scheduler-config)
+          compute-cluster->scheduling-capacity (pc/map-from-keys #(cc/max-launchable % pool-name) compute-clusters-for-pool)
+          available-scheduling-capacity (reduce + (vals compute-cluster->scheduling-capacity))
+          ; We can avoid some expensive calculations (e.g. quotas) for jobs
+          ; that ultimately will not be scheduled this cycle, if we clamp
+          ; max considerable to the current available scheduling capacity.
+          max-considerable (min max-total-considerable available-scheduling-capacity)]
+      (log-structured/info "Cycle available scheduling capacity"
+                           {:available-scheduling-capacity available-scheduling-capacity 
+                            :max-considerable max-considerable :pool pool-name})
+      
+      ; A minimum threshold prevents incurring cycle overhead for a small number
+      ; of jobs that could be scheduled this cycle, due to low available scheduling capacity.
+      (if (> available-scheduling-capacity minimum-scheduling-capacity-threshold)
+        (let [db (db conn)
+              pending-jobs (get @pool-name->pending-jobs-atom pool-name)
+              user->usage (tracing/with-span [s {:name "scheduler.kubernetes-handler.resolve-user-to-usage-future"
+                                                 :tags {:pool pool-name :component tracing-component-tag}}]
+                            @user->usage-future)
+              considerable-jobs (pending-jobs->considerable-jobs db pending-jobs user->quota user->usage max-considerable pool-name)
+              considerable-job-uuids (set (map :job/uuid considerable-jobs))]
+          (log-structured/info "Considering jobs" {:pool pool-name :number-considered-jobs (count considerable-job-uuids)
+                                                   :number-pending-jobs (count pending-jobs)})
+          (if (seq considerable-jobs)
+            (do
+              (swap! pool-name->pending-jobs-atom
```

Comment:
Worth a small comment: 
`remove the considerable jobs from consideration from matching, until we do a rank cycle again.`

Reply:
done